### PR TITLE
v1.6.0 Fix most features.

### PR DESCRIPTION
**Pull request by gooin**
-   [x] New HttpClient class
-   [x] Login and get user token
-   [x] Garmin URLs works with `garmin.cn` and `garmin.com`
-   [x] Auto refresh Ouath2 token
-   [x] Oauth1,Oauth2 token import and export.
-   [x] Download Activity, countActivities, getActivities, getActivity, getUserProfile, getUserSettings
-   [x] Upload Activity, delete Activity
-   [ ] Implementation of other methods, such as Badge,Workout,Gear etc
-   [ ] Handle MFA
-   [x] Handle Account locked
-   [ ] Unit test
-   [ ] Listeners

### CODE REVIEW

**Comment by Abasz** [comment]
Is there a specific reason that you are still using the `/proxy/` path on the connect api? 

Actually, for some reasons I got errors on that and moved to `connectapi.garmin.com` without the `/proxy/`. I checked and I think this is used by Garth as well.

So for instance, activity list service: `https://connectapi.garmin.com/activitylist-service/activities/search/activities`
or download service: `https://connectapi.garmin.com/download-service/files/activity/`

**Comment by gooin** [comment]
> > > Is there a specific reason that you are still using the `/proxy/` path on the connect api?
> > > Actually, for some reasons I got errors on that and moved to `connectapi.garmin.com` without the `/proxy/`. I checked and I think this is used by Garth as well.
> > > So for instance, activity list service: `https://connectapi.garmin.com/activitylist-service/activities/search/activities` or download service: `https://connectapi.garmin.com/download-service/files/activity/`
> > 
> > 
> > I use `UrlClass.ts` not `url.ts`
> 
> you are right... sorry I missed that.

I Will delete it later.😂

**Comment by gooin** [comment]
> > > Is there a specific reason that you are still using the `/proxy/` path on the connect api?
> > > Actually, for some reasons I got errors on that and moved to `connectapi.garmin.com` without the `/proxy/`. I checked and I think this is used by Garth as well.
> > > So for instance, activity list service: `https://connectapi.garmin.com/activitylist-service/activities/search/activities` or download service: `https://connectapi.garmin.com/download-service/files/activity/`
> > 
> > 
> > I use `UrlClass.ts` not `url.ts`
> 
> you are right... sorry I missed that.

@Abasz removed...

**Comment by kspetrov** [comment]
@gooin Hi! What about workout methods? Or this is not in your plan?

**Comment by gooin** [comment]
Yes, I'm working on this.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by kspetrov** [comment]
Perfect! :) Thanks a lot!!!

**Comment by gooin** [comment]
> Perfect! :) Thanks a lot!!!

Latest commit added workout methods. See https://github.com/Pythe1337N/garmin-connect/pull/56/commits/901a70f2b217e85454d596baf9d640023a34aa34

**Comment by pauly** [comment]
Is the whole interface changing here? The original examples aren't working, neither is my code if I switch to this branch.
I see the change from 

```
await GCClient.login(process.env.GUSER, process.env.GPASS)
const userProfile = await GCClient.getUserProfile()
const stepData = await GCClient.getSteps(new Date(date))
```
now there is no getSteps()

**Comment by gooin** [comment]
> Is the whole interface changing here? The original examples aren't working, neither is my code if I switch to this branch. I see the change from
> 
> ```
> await GCClient.login(process.env.GUSER, process.env.GPASS)
> const userProfile = await GCClient.getUserProfile()
> const stepData = await GCClient.getSteps(new Date(date))
> ```
> 
> now there is no getSteps()

steps, sleep, weight, gear are not implemented yet, see `GarminConnect.ts`

File: package.json [hunk 1]
```diff
@@ -1,12 +1,14 @@
 {
     "name": "garmin-connect",
-    "version": "1.5.0",
+    "version": "1.6.5",
```
Comment (Pythe1337N) [inline package.json:3]: Please do not bump version numbers, this will be done along with a new release

**Review by Pythe1337N — CHANGES_REQUESTED** [review]

**Comment by app4g** [comment]
Based on the above (first post) the gear link/unlink is still WIP correct?

https://github.com/Pythe1337N/garmin-connect/pull/46

I'm trying to give it a go but am getting Status  403 Forbidden

```PUT https://connectapi.garmin.com/gear-service/gear/link/(gearUUID)/activity/(activityId)```

```
{
  "message": null,
  "error": "ForbiddenException"
}
```

this used to work using the previous Web authentication style.

**Comment by matin** [comment]
@gooin well done! 👏🏽

**Pull request merged**